CMSgov/easi-app
# [task: EASI-3933] Refactor and Update and LCID expiration alert timing

# EASI-3933


## Changes and Description

- Refactored logic for calculating when to alert a given intake into a function (`shouldSendAlertForIntake`). This helped reduce some code repetition, as well as allowed for more seamless testing.
- Updated LCID alerting logic to now alert on 120, 60, and 14 days (effectively adding the 120-day mark to the list)

## How to test this change

- `scripts/dev up`
- Create a System Intake with an LCID that expires 130 days from now
  - Restart the BE docker container (as this will cause the LCID alert code to restart)
  - Assert no alert was sent in mailcatcher
- Update the LCID expiration to be 110 days from now
  - Restart the BE docker container (as this will cause the LCID alert code to restart)
  - Assert that an alert was sent in mailcatcher
- Restart the BE container without updating the expiration
  - Assert no alert was sent in mailcatcher (one was already sent for 120 days). You can repeat this by updating the LCID expiration to any other date between 120 and 60 days away, like 90 or 80.


You should be able to effectively move the LCID expiration around as much as you'd like, and just assert that you only get an alert when you cross one of the thresholds defined in `alertsInAdvance` (120, 60, or 14 days)

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
